elyra/__init__.py

```diff
@@ -35,14 +36,16 @@ def load_jupyter_server_extension(nb_server_app):
     web_app = nb_server_app.web_app
     host_pattern = '.*$'
     web_app.add_handlers(host_pattern, [
-        (url_path_join(web_app.settings['base_url'], r'/api/{}'.format(JsonSpecHandler.get_resource_metadata()[0])),
-         JsonSpecHandler),
         (url_path_join(web_app.settings['base_url'], r'/api/{}'.format(YamlSpecHandler.get_resource_metadata()[0])),
          YamlSpecHandler),
         (url_path_join(web_app.settings['base_url'], r'/api/scheduler'), SchedulerHandler),
         (url_path_join(web_app.settings['base_url'], r'/api/metadata/%s' % (namespace_regex)), MetadataHandler),
         (url_path_join(web_app.settings['base_url'], r'/api/metadata/%s/%s' % (namespace_regex, resource_regex)),
-         MetadataNamespaceHandler),
+         MetadataResourceHandler),
+        (url_path_join(web_app.settings['base_url'], r'/api/schema/%s' % (namespace_regex)), SchemaHandler),
+        (url_path_join(web_app.settings['base_url'], r'/api/schema/%s/%s' % (namespace_regex, resource_regex)),
+         SchemaResourceHandler),
+        (url_path_join(web_app.settings['base_url'], r'/api/namespace'), NamespaceHandler),
```

Comment:
should these new apis be relative to metadata?  like /api/metadata/schema?

Reply:
They are currently, but that's not necessarily the case down the road.  However, the primary reason for the change was in talking with @vabarbosa we both felt the previous endpoint `/api/metadata_schema` was too long and using something like `/api/metadata/schema/{namespace}` would be ambiguous with the current `/api/metadata/{namespace}/{resource}` endpoint, resulting in an attempt to return the metadata instance named {resource} of namespace 'schema' instead of the schemas in namespace {resource}. I had considered using that endpoint approach initially. :smile:

Reply:
Can't the metadata handler check if `path` is schema and redirect to `MetadataSchema handler`?

Reply:
I suppose it could, although I suspect just "redirecting" to another handler is much more involved than that since the entire request context also needs to be conveyed. However, I don't think that's correct to do anyway since, although completely hypothetical, it prevents the ability to ever have a namespace of `schema` - per my previous comment.